src/components/dropdownSearch/DropdownSearchReducers.tsx

```diff
@@ -44,20 +48,88 @@ export const getNextIndexPosition = (array: any[], item: any, key: number): numb
   return index;
 };
 
-export const getOptionsFiltered = (state: IDropdownSearchState, filterText?: string): IDropdownOption[] => {
-  const currentFilterText: string = filterText || state.filterText;
-  if (_.isEmpty(currentFilterText)) {
-    return state.options;
+export const deselectOption = (options: IDropdownOption[], value: string): IDropdownOption[] => {
+  const nextOptions: IDropdownOption[] = deepClone(options);
+  const selectedOption = _.find(nextOptions, { value });
+  if (selectedOption) {
+    if (selectedOption.custom) {
+      nextOptions.splice(nextOptions.indexOf(selectedOption), 1);
+    } else {
+      selectedOption.selected = false;
+      selectedOption.hidden = false;
+    }
   }
-  return _.filter(state.options,
+  return nextOptions;
+};
+
+export const deselectLastSelectedOption = (options: IDropdownOption[]): IDropdownOption[] => {
+  const lastSelectedOption: IDropdownOption = _.find(options.slice().reverse(), { selected: true });
+  return lastSelectedOption
+    ? deselectOption(options, lastSelectedOption.value)
+    : deepClone(options);
+};
+
+export const deselectAllOptions = (options: IDropdownOption[]): IDropdownOption[] => {
+  let nextOptions: IDropdownOption[] = [];
+  _.each(options, (option: IDropdownOption) => {
+    if (!option.custom) {
+      const nextOption = deepClone(option);
+      nextOptions.push({ ...nextOption, selected: false, hidden: false });
+    }
+  });
+
+  return nextOptions;
+};
+
+export const addUniqueSelectedOption = (options: IDropdownOption[], value: string): IDropdownOption[] => {
+  const sameValueDoesNotExist = _.findWhere(options, { value }) === undefined;
+
+  if (sameValueDoesNotExist) {
+    const nextOptions: IDropdownOption[] = deepClone(options);
+    nextOptions.push({
+      value,
+      selected: true,
+      custom: true,
+    });
+    return nextOptions;
+  }
+  return deepClone(options);
+};
+
+export const getDisplayedOptions = (options: IDropdownOption[]): IDropdownOption[] => {
+  return _.reject(options, (option) => {
+    return option.custom || option.hidden;
+  });
+};
+
+export const getFilteredOptions = (state: IDropdownSearchState, filterText?: string) => {
+  const currentFilterText: string = filterText || state.filterText || '';
+  return _.filter(getDisplayedOptions(state.options),
     (option: IDropdownOption) => {
       const value = option.displayValue || option.value;
```

Comment:
const displayValue = option.displayValue || option.value; 

what you are handling here is the value displayed in the ui